core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -372,16 +372,105 @@ ClientRequestContext newDerivedContext(RequestId id, @Nullable HttpRequest req,
      * Sets the amount of time allowed until receiving the {@link Response} completely
      * since the transfer of the {@link Response} started. This value is initially set from
      * {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * ctx.setResponseTimeoutMillis(1000);
+     * assert ctx.responseTimeoutMillis() == 1000;
+     * // Do something...
+     * ctx.setResponseTimeoutMillis(2000);
+     * assert ctx.responseTimeoutMillis() == 2000;
+     * }</pre>
+     *
+     * @deprecated Use {@link #adjustResponseTimeoutMillis(long)} or {@link #resetResponseTimeoutMillis(long)}
      */
+    @Deprecated
     void setResponseTimeoutMillis(long responseTimeoutMillis);
 
     /**
      * Sets the amount of time allowed until receiving the {@link Response} completely
      * since the transfer of the {@link Response} started. This value is initially set from
      * {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * ctx.setResponseTimeoutMillis(1000);
+     * assert ctx.responseTimeoutMillis() == 1000;
+     * // Do something...
+     * ctx.setResponseTimeoutMillis(2000);
+     * assert ctx.responseTimeoutMillis() == 2000;
+     * }</pre>
+     * @deprecated Use {@link #adjustResponseTimeout(Duration)} or {@link #resetResponseTimeout(Duration)}
      */
+    @Deprecated
     void setResponseTimeout(Duration responseTimeout);
 
+    /**
+     * Adjusts the existing response timeout allowed by the specified {@code adjustmentMillis} until
+     * receiving the {@link Response} completely since the transfer of the {@link Response} started.
+     * This value is initially set from {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * long oldResponseTimeoutMillis = ctx.responseTimeoutMillis();
+     * ctx.adjustResponseTimeoutMillis(1000);
+     * assert ctx.responseTimeoutMillis() == oldResponseTimeoutMillis + 1000;
+     * }</pre>
+     */
+    void adjustResponseTimeoutMillis(long adjustmentMillis);
```

Comment:
extend is probably easier to understand than adjust

Reply:
Also, I want to repropose once more making the API accept an absolute deadline instead.

```java
ctx.setResponseDeadline(ctx.responseDeadline() + 1000); // Adjust isn't that much more code and seems more readable being just simple get/set, no new concept of "adjust"

ctx.setResponseDeadline(midnight); // Absolute deadline with current API is very obtuse. 

ctx.setResponseDeadlineAfter(1000); // Convenient but equivalent to ctx.setResponseDeadline(Duration.ofNanos(System.nanoTime() + 1000000));
```

Reply:
> extend is probably easier to understand than adjust

I thought it is not just extending the exising timeout. The timeout could be set to a negative value to reduce the existing timeout.

Reply:
Yeah it's definitely a valid concern but I think it's similar to `add` type methods accepting a negative value and actually subtracting instead of adding so still natural IMO. On the flip side, adjust is a very generic word so it's hard to understand what it actually does from just the method name.

Reply:
> Also, I want to repropose once more making the API accept an absolute deadline instead.

It looks good to me. Let me summarize our new timeout API spec based on your proposal. 😀